js/background.js

```diff
@@ -746,41 +772,50 @@
         identityKeys.privKey
       );
 
-      // Constructing a "create group" message
-      const proto = new textsecure.protobuf.DataMessage();
-
-      const groupUpdate = new textsecure.protobuf.MediumGroupUpdate();
+      const primary = window.storage.get('primaryDevicePubKey');
 
-      groupUpdate.groupId = groupId;
-      groupUpdate.groupSecretKey = groupSecretKeyHex;
-      groupUpdate.senderKey = senderKey;
-      groupUpdate.members = [ourIdentity, ...members];
-      groupUpdate.groupName = groupName;
-      proto.mediumGroupUpdate = groupUpdate;
+      const allMembers = [primary, ...members];
 
       await window.Signal.Data.createOrUpdateIdentityKey({
         id: groupId,
         secretKey: groupSecretKeyHex,
       });
 
-      const convo = await window.ConversationController.getOrCreateAndWait(
+      const ev = new Event('group');
```

Comment:
The difference should be only when the event is emited. I simplified it since it is passed directly as a collection of properties. I could simplify in other places too.

Reply:
Updated the other places too.